keadm/cmd/keadm/app/cmd/util/common.go

```diff
@@ -386,31 +386,34 @@ func installKubeEdge(componentType types.ComponentType, arch string, version sem
 		When installing edgecore, if the version is >= 1.1,
 		download the edgecore.service file from the KubeEdge/build/tools/ and place it in /etc/kubeedge/ acc.
 	*/
-	if componentType == types.EdgeCore {
-		strippedVersion := fmt.Sprintf("%d.%d", version.Major, version.Minor)
-
-		//	No need to download if the version is less than 1.1 (or 1.1.0)
-		if version.GE(semver.MustParse("1.1.0")) {
-			try := 0
-
-			edgecoreServiceFileName := "edgecore.service"
+	strippedVersion := fmt.Sprintf("%d.%d", version.Major, version.Minor)
 
+	//	No need to download if the version is less than 1.1 (or 1.1.0)
+	if hasSystemd() && version.GE(semver.MustParse("1.1.0")) {
+		var urlForServiceFile string
+		var ServiceFileName string
+		if componentType == types.EdgeCore {
+			ServiceFileName = "edgecore.service"
 			if version.EQ(semver.MustParse("1.1.0")) {
-				edgecoreServiceFileName = "edge.service"
+				ServiceFileName = "edge.service"
```

Comment:
Unit as `edgecore.service` is better.

Reply:
It is a history question, in version v1.1.0, service file name called `edge.service` as https://github.com/kubeedge/kubeedge/blob/release-1.1/build/tools/edge.service

Reply:
Get.